apps/transport/lib/jobs/dataset_quality_score.ex

```diff
@@ -0,0 +1,281 @@
+
+  @spec resource_freshness(map) ::
+          %{
+            :format => binary,
+            :freshness => float | nil,
+            :resource_id => integer,
+            :raw_measure => any
+          }
+  def resource_freshness(%{format: "GTFS", id: resource_id}) do
+    resource_id
+    |> DB.MultiValidation.resource_latest_validation(Transport.Validators.GTFSTransport)
+    |> case do
+      %{metadata: %{metadata: %{"start_date" => start_date, "end_date" => end_date}}}
+      when not is_nil(start_date) and not is_nil(end_date) ->
+        start_date = Date.from_iso8601!(start_date)
+        end_date = Date.from_iso8601!(end_date)
+
+        today = Date.utc_today()
+
+        freshness =
+          if Date.compare(start_date, today) != :gt and Date.compare(today, end_date) != :gt,
+            do: 1.0,
+            else: 0.0
+
+        %{
+          freshness: freshness,
+          raw_measure: %{start_date: start_date, end_date: end_date}
+        }
+
+      _ ->
+        %{
+          freshness: nil,
+          raw_measure: nil
+        }
+    end
+    |> Map.merge(%{resource_id: resource_id, format: "GTFS"})
+  end
+
+  def resource_freshness(%{format: "gbfs", id: resource_id}) do
+    freshness =
+      resource_id
+      |> resource_last_metadata_from_today()
+      |> case do
+        %{metadata: %{"feed_timestamp_delay" => feed_timestamp_delay}} ->
+          %{
+            freshness: gbfs_feed_freshness(feed_timestamp_delay),
+            raw_measure: feed_timestamp_delay
+          }
+
+        _ ->
+          %{
+            freshness: nil,
+            raw_measure: nil
+          }
+      end
+      |> Map.merge(%{resource_id: resource_id, format: "gbfs"})
+
+    freshness
+  end
+
+  def resource_freshness(%{format: "gtfs-rt", id: resource_id}) do
+    freshness =
+      resource_id
+      |> resource_last_metadata_from_today()
+      |> case do
+        %{metadata: %{"feed_timestamp_delay" => feed_timestamp_delay}} ->
+          %{
+            freshness: gtfs_rt_feed_freshness(feed_timestamp_delay),
+            raw_measure: feed_timestamp_delay
+          }
+
+        _ ->
+          %{
+            freshness: nil,
+            raw_measure: nil
+          }
+      end
+      |> Map.merge(%{resource_id: resource_id, format: "gtfs-rt"})
+
+    freshness
+  end
+
+  def resource_freshness(%DB.Resource{}), do: nil
+
+  @doc """
+  5 minutes is the max delay allowed for GBFS
+  https://github.com/MobilityData/gbfs/blob/master/gbfs.md#data-latency
+  """
+  def gbfs_max_timestamp_delay, do: 5 * 60
+
+  @doc """
+  gives a feed a freshness score, based on observed feed_timestamp_delay
+
+  iex> gbfs_feed_freshness(0)
+  1.0
+  iex> gbfs_feed_freshness(1000)
+  0.0
+  """
+  def gbfs_feed_freshness(feed_timestamp_delay) do
+    if feed_timestamp_delay < gbfs_max_timestamp_delay(), do: 1.0, else: 0.0
+  end
+
+  @doc """
+  we allow a 5 minutes delay for GTFS realtime feeds
+  """
+  def gtfs_rt_max_timestamp_delay, do: 5 * 60
```

Comment:
On va mettre des warnings à la validation dès que c'est plus de 30-60s https://github.com/CUTR-at-USF/gtfs-realtime-validator/blob/master/RULES.md#table-of-warnings

Reply:
Je trouve ça confortable de commencer laxiste, puis de resserrer un peu nos exigences par la suite...